models/classes/ItemPreviewer.php

```diff
@@ -0,0 +1,279 @@
+    }
+
+    /**
+     * @return string
+     * @throws \common_Exception
+     * @throws \common_exception_Error
+     * @throws \common_exception_InconsistentData
+     */
+    public function getBaseUrl()
+    {
+        return $this->getItemPublicDir()->getPublicAccessUrl() . $this->getUserLanguage() . '/';
+    }
+
+    /**
+     * @return string
+     * @throws \common_exception_Error
+     */
+    private function getUserLanguage()
+    {
+        if (is_null($this->userDataLang)) {
+            /** @var ResultServerService $resultServerService */
+            $resultServerService = $this->getServiceLocator()->get(ResultServerService::SERVICE_ID);
+            /** @var \taoResultServer_models_classes_ReadableResultStorage $implementation */
+            $implementation = $resultServerService->getResultStorage($this->delivery->getUri());
+
+            $testTaker = new \core_kernel_users_GenerisUser(new \core_kernel_classes_Resource($implementation->getTestTaker($this->resultId)));
+            $lang = $testTaker->getPropertyValues(\oat\generis\model\GenerisRdf::PROPERTY_USER_DEFLG);
+
+            $this->userDataLang = empty($lang) ? DEFAULT_LANG : (string) current($lang);
+        }
+
+        return $this->userDataLang;
+    }
+
+    /**
+     * @return \tao_models_classes_service_StorageDirectory
+     * @throws \common_exception_InconsistentData
+     */
+    private function getItemPublicDir()
+    {
+        if (is_null($this->itemPublicDir)) {
+            $this->itemPublicDir = $this->fileStorage->getDirectoryById($this->getItemPublicHref());
+        }
+
+        return $this->itemPublicDir;
+    }
+
+    /**
+     * @return \tao_models_classes_service_StorageDirectory
+     * @throws \common_exception_InconsistentData
+     */
+    private function getItemPrivateDir()
+    {
+        if (is_null($this->itemPrivateDir)) {
+            $this->itemPrivateDir = $this->fileStorage->getDirectoryById($this->getItemPrivateHref());
+        }
+
+        return $this->itemPrivateDir;
+    }
+
+    /**
+     * @return string
+     * @throws \common_exception_InconsistentData
+     */
+    public function getItemUri()
+    {
+        if (empty($this->itemHrefs)) {
+            $this->loadItemHrefs();
+        }
+
+        return $this->itemHrefs[0];
+    }
+
+    /**
+     * @return string
+     * @throws \common_exception_InconsistentData
+     */
+    private function getItemPublicHref()
+    {
+        if (empty($this->itemHrefs)) {
+            $this->loadItemHrefs();
+        }
+
+        return $this->itemHrefs[1];
+    }
+
+    /**
+     * @return string
+     * @throws \common_exception_InconsistentData
+     */
+    private function getItemPrivateHref()
+    {
+        if (empty($this->itemHrefs)) {
+            $this->loadItemHrefs();
+        }
+
+        return $this->itemHrefs[2];
+    }
+
+    private function loadItemHrefs()
+    {
+        $runtimeService = $this->getServiceLocator()->get(RuntimeService::SERVICE_ID);
```

Comment:
The link with the delivery needs to be validated in terms of architecture

Reply:
Sorry, what you mean?

Reply:
The fact the implementation rely on the DeliveryContainer used. If @joel can have a look to check if it's ok